apps/src/templates/curriculumCatalog/CurriculumCatalog.jsx

```diff
@@ -7,48 +7,111 @@ import HeaderBanner from '../HeaderBanner';
 import CurriculumCatalogCard from '@cdo/apps/templates/curriculumCatalog/CurriculumCatalogCard';
+import CheckboxDropdown from '../CheckboxDropdown';
 
-const CurriculumCatalog = ({curriculaData, isEnglish}) => (
-  <>
-    <HeaderBanner
-      headingText={i18n.curriculumCatalogHeaderTitle()}
-      subHeadingText={i18n.curriculumCatalogHeaderSubtitle()}
-      short={false}
-      backgroundUrl={CourseCatalogBannerBackground}
-      imageUrl={CourseCatalogIllustration01}
-    />
-    <div className={style.catalogContentContainer}>
-      <div className={style.catalogContent}>
-        {/*TODO [MEG]: calculate and pass in duration and translated from backend */}
-        {curriculaData
-          .filter(curriculum => !!curriculum.grade_levels)
-          .map(
-            ({
-              key,
-              image,
-              display_name,
-              grade_levels,
-              school_subject,
-              cs_topic
-            }) => (
-              <CurriculumCatalogCard
-                key={key}
-                courseDisplayName={display_name}
-                imageSrc={image || undefined}
-                duration={'school_year'} // TODO [MEG] actually pass in this data
-                gradesArray={grade_levels.split(',')}
-                subjects={school_subject?.split(',')}
-                topics={cs_topic?.split(',')}
-                isTranslated={false} // TODO [MEG]: actually pass in this data
-                isEnglish={isEnglish}
-              />
-            )
-          )}
-        ;
+const CurriculumCatalog = ({curriculaData, isEnglish}) => {
+  const filterTypes = [
+    {name: 'course', label: 'Course', options: ['CSA', 'CSD', 'CSP', 'CSF']},
+    {name: 'color', label: 'Color', options: ['Red', 'Blue', 'Green']}
+  ];
+
+  const getClearedFilters = () => {
+    let filters = {};
+    filterTypes.forEach(filter => {
+      filters[filter.name] = [];
+    });
+    return filters;
+  };
+
+  const [appliedFilters, setAppliedFilters] = useState(getClearedFilters());
+
+  const handleSelect = (event, filterName) => {
+    const value = event.target.value;
+    const isChecked = event.target.checked;
+
+    let newFilters = appliedFilters;
+    if (isChecked) {
+      //Add checked item into applied filters
+      newFilters[filterName] = [...appliedFilters[filterName], value];
+      setAppliedFilters(newFilters);
+    } else {
+      //Remove unchecked item from applied filters
+      newFilters[filterName] = appliedFilters[filterName].filter(
+        item => item !== value
+      );
+      setAppliedFilters(newFilters);
+    }
+  };
+
+  const handleClear = () => {
+    filterTypes.forEach(filterType => {
+      filterType.options.forEach(option => {
+        document.getElementById(
+          `${filterType.name}-${option}-check`
+        ).checked = false;
+      });
+    });
+    setAppliedFilters(getClearedFilters());
+  };
```

Comment:
I'm not stoked about using `getElementById` then changing the property as the strategy for clearing the checkboxes, but with trying to split up the `CheckboxDropdown` component and the rest of it (so that the CurriculumCatalog didn't get too crowded and hard to read) this was what I was able to get working. I'm super open to pushback if this should be improved or if there are other ideas on how to split up CheckboxDropdown and the rest such that I can maintain this functionality!

Reply:
hey Turner, I'm not sure I follow exactly how this is working, but the getElementById is definitely a red flag. I would naively expect:
1. the list (or lists) of currently selected filters to be part of the react state on the CurriculumCatalog component
2. handleClear to set these lists to empty
3. the`CheckboxDropdown` uses callbacks (passed in as props by curriculum catalog) to add or remove from the current list of filter items
4. the `CheckboxDropdown` is a [controlled component](https://legacy.reactjs.org/docs/forms.html#controlled-components)

does that help? happy to go through this in more detail.

> with trying to split up the CheckboxDropdown component and the rest of it (so that the CurriculumCatalog didn't get too crowded and hard to read)

I am wondering if this is the source of challenge here -- what are you trying to keep out of CurriculumCatalog exactly?

Reply:
`FormController` (the one in `form_components_func`) is a (very complex) controlled component, as is `PetitionForm` if that's helpful